### PR TITLE
loading spinner added

### PR DESCRIPTION
**Pull request by samikuikka**
loading spinner that is shown when prompts are send to the backend.

### CODE REVIEW

**Review by itulkki — APPROVED** [review]
looks good

**Review by KorpelainenTommi — APPROVED** [review]
This works, but perhaps the loading should not be implemented as a global page load? (Load spinner blocking everything) The user has no way to abort loading, and has no agency during a load.

In the future, this could be replaced with a system that allows content panels (or even individual promptboxes) to be in a loading state independently. This way, the user could switch to another panel to work while they wait for one content panel to receive data.

TL;DR approved these changes, but this should be changed later (probably when requests get changed to a websocket based solution)
